### DISCUSSION TITLE
Salesforce objects missing fields

### DISCUSSION THREAD

**Discussion by ghost** [airbytehq/airbyte#45662] [Connector Questions]
When pulling custom Salesforce objects with Airbyte, some of the fields are missing. I've deployed Airbyte with docker compose, and am using airbyte/source-salesforce:2.5.24.

I've reset the stream, and the sync is full refresh overwrite.

What could cause fields to be missing?

**Comment by nataliekwong** [comment]
Hi there, 

- Do you see the column in the schema tab of your connection?
- If you do see the columns, are they not showing up in the destination raw or final tables? What is your destination? 

Most commonly, if the columns do not appear in the schema tab, it is because the column is hidden for the user that authenticated to Salesforce.

**Reply by ghost** [reply]
Thanks for the quick response! I will check this and get back to you.

**Comment by ghost** [comment]
Using SOQL, I'm able to query the object, and it seems like I can see the field when using my personal credentials.

However, I don't see the field in the schema tab of Airbyte.

For the Salesforce source in Airbyte, I provide the Client ID, Client Secret, and Refresh Token.

If I understanding correctly, the Client ID might not have access to the column that's missing?

**Comment by nataliekwong** [comment]
Interesting, good to know that you can query via SOQL. That would indicate it is not a permissions issue with the field being accessible to the API.

- Are these custom or standard fields? Is the object a standard or custom object?
- Is it possible the display name and API name of the fields have diverged and you are looking for the display name?

**Reply by ghost** [reply]
I was able to query the object with SOQL using my personal credentials.
The credentials Airbyte uses are different.
It tried to use the airbyte credentials to query with SOQL in Python, but couldn't.

These are custom objects and I believe a mix of custom and standard fields. For example, I believe UserRecordAccessId to be a standard field.

No that isn't the issue. I cannot see the field by any name in the schema inspector.

**Reply by nataliekwong** [reply]
Oh, got it. Then it might be indeed a permissions issue. I did some looking around to see how you might check if fields are accessible to the user: 

- You can check on an individual field to verify it's accessible for the profile & record type you're looking for.(https://help.salesforce.com/s/articleView?id=sf.checking_field_accessibility_for_a_particular_field.htm&type=5)
- If your authenticated user has a permission set assigned to it, you can check what the settings are for each object and field. The permission set should have read access to the field. (https://help.salesforce.com/s/articleView?id=sf.users_profiles_fls.htm&type=5)

**Reply by gmorrell** [reply]
Just adding somethig to this i am having the same issue

I tried syncing some but not all fields mix of custom and standard fields but 1 seems to be missing

can see it on the schema tab in airbyte, in the raw tables in the internal tables just not in the final table.

This is a fairly new column could it be somethign that airbyte is using a sample and all records are null so its excluded? or does it not do that... ive tried reyncing deleting the column readding it but no luck?

seems that i have the access as the integration user?

[a comment by an automated account is not shown]
